intel_extension_for_transformers/transformers/modeling/modeling_gaudi/models/llama/modeling_llama.py

```diff
@@ -329,6 +329,7 @@ def pre_attn_forward(
         flash_attention_recompute: Optional[bool] = False,
         flash_attention_causal_mask: Optional[bool] = False,
         cache_idx: int = None,
+        cache_prune_num: int = 0,
```

Comment:
how about more straightforward like ```cache_remained_size```

Reply:
what is the relationship between this and ```attention_sink```

Reply:
I tend to keep it. `attention_sink_size` + `attention_sink_window_size` = `kv_cache_len`. qk matmul will always be [1, head_num, kv_seq_len, kv_seq_len] for static shapes (=`cache_remained_size` you said here). `cache_prune_num` will guide `KVCache` class how to reorder cache but its size keeps same.

Reply:
relationship is: if total_tokens > kv_cache_len, will dorp `cache_prune_num` tokens. `attention_sink` has its own drop ways (from "middle").

Reply:
how to cal `cache_prune_num` is from outside because the modeling (graph) can't use dynamic flow. Otherwise, I will hide it.